src/helpers/databaseHelpers.ts

```diff
@@ -0,0 +1,113 @@
+import supabase from '@/utils/supabase-client';
+
+interface ScoreRecord {
+  address: string;
+  score?: number;
+  meta?: any;
+  version?: number;
+  attestation?: string;
+  ipfs_hash?: string;
+  receipt?: string;
+}
+
+export async function updateScoreRecord(record: ScoreRecord) {
+  try {
+    const { address, ...data } = record;
+
+    // Check if a record with the given address already exists
+    const { data: existingData, error: selectError } = await supabase
```

Comment:
We will want to display score over time in the future, so let's make each measurement a separate record. We already account for this in the percentiles calculation.